origins_social_sharing/origins_social_sharing.module

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * @file
+ * Contains origins_social_sharing.module.
+ */
+
+use Drupal\Core\Url;
+
+/**
+ * Implements hook_theme().
+ */
+function origins_social_sharing_theme() {
+  return [
+    'origins_social_sharing' => [
+      'variables' => [
+        'fb_link' => NULL,
+        'tw_link' => NULL,
+        'mail_link' => NULL,
+      ],
+    ],
+  ];
+}
+
+/**
+ * Implements hook_preprocess_block() for block content.
+ */
+function origins_social_sharing_preprocess_block(&$variables) {
+  // Look for the 'social sharing' block.
+  if ($variables['plugin_id'] == 'origins_social_sharing') {
+    // Get current URL so that we can pass it to Facebook.
+    $url = Url::fromRoute('<current>', [], ['absolute' => 'true']);
+    $fb_link = 'https://www.facebook.com/sharer/sharer.php?u=' . rawurlencode($url->toString()) . '&amp;src=sdkpreparse';
+    $variables['content']['#fb_link'] = $fb_link;
+    // Get current node and extract title.
+    $node = \Drupal::routeMatch()->getParameter('node');
+    $title = '';
+    if (!empty($node) && is_object($node)) {
+      $title = $node->getTitle();
+    }
+    // Build Twitter link.
+    $tw_link = 'https://twitter.com/intent/tweet?text=' . rawurlencode($title) . ' ' . rawurlencode($url->toString());
+    $variables['content']['#tw_link'] = $tw_link;
+    // Build mailto link.
+    $mail_link = "mailto:?subject=" . t("Shared from @host", ['@host' => $_SERVER['HTTP_HOST']]);
```

Comment:
This will work, but I seem to recall Drupal wanting to replace references to superglobal values with their own internal abstractions of, in the case of Drupal 8, the Symfony request objects: https://www.drupal.org/node/2150267

I think the motivation behind this is that sometimes underlying system/service environment variables can vary whereas the Symfony objects will not, or will vary far less.

Reply:
Yes, good spot